### PR TITLE
CB-197 - create the extraordinary installments modal

### PR DESCRIPTION
**Pull request by diegoAndres01**
[Card Jira](https://inube.atlassian.net/browse/CB-197)

image

### CODE REVIEW

**Review by Angelbt12 — APPROVED** [review]

File: src/components/data/TableExtraordinaryPayment/Actions/styles.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+const StyledLi = styled.li`
+  list-style: none;
+  display: flex;
+  align-items: center;
+  padding: 6px 0px;
+  cursor: pointer;
+`;
+
+const StyledActions = styled.div`
+  background-color: ${({ theme }) =>
```
Comment (carin91) [inline src/components/data/TableExtraordinaryPayment/Actions/styles.ts:32]: relocate the css prop with inputs at the end of the component

File: src/components/data/TableExtraordinaryPayment/Actions/type.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+interface IAction {
+  icon: React.ReactNode;
+  appearance: keyof typeof IAppearances;
+  label: string;
+  onClick?: () => void;
+}
+
+export const IAppearances = {
```
Comment (carin91) [inline src/components/data/TableExtraordinaryPayment/Actions/type.ts:8]: This appearances are general in the design system, verify that if the general enum or types file includes this apperances or import it from the design system (if available) or create it in the general enum file

File: src/components/data/TableExtraordinaryPayment/Detail/index.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      <Icon
+        icon={<MdOutlineMoreVert />}
+        size="12px"
+        cursorHover
+        appearance="primary"
+        onClick={() => setModalOpen(true)}
+        shape="circle"
+        variant="filled"
+        />
+      {ModalOpen && <ActionModal onClose={() => setModalOpen(false)}  onClickDetails={onClickDetails} onClickEdit={onClickEdit} onClickEliminate={onClickEliminate} />}
```
Comment (carin91) [inline src/components/data/TableExtraordinaryPayment/Detail/index.tsx:27]: Verify that you install prettier

**Review by carin91 — APPROVED** [review]

File: src/components/data/TableExtraordinaryPayment/index.tsx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+import { useMediaQuery } from "@inubekit/hooks";
+import { IRowExtraordinaryPayment } from "@components/modals/ExtraordinaryPaymentModal/types"
+
+import {
+  headersTableExtraordinaryPayment,
+  rowsVisbleMobile,
+  rowsActions,
+} from "./config";
+
+export interface TableExtraordinaryPaymentProps {
```
[the remark this answers is not in the document]
Comment (diegoAndres01) [reply src/components/data/TableExtraordinaryPayment/index.tsx:23]: resolved

File: src/components/modals/ExtraordinaryPaymentModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
+import { createPortal } from "react-dom";
```
[the remark this answers is not in the document]
Comment (diegoAndres01) [reply src/components/modals/ExtraordinaryPaymentModal/index.tsx:1]: resolved

File: src/components/data/TableExtraordinaryPayment/Actions/index.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  onClickEliminate: () => void;
+}
+
+export function ActionModal(props: ActionModalProps) {
+  const { onClose, onClickDetails, onClickEdit, onClickEliminate } = props;
+  const actions: IAction[] = [
+    {
+      icon: <MdOutlineRemoveRedEye />,
+      appearance: IAppearances.dark,
+      label: "Ver detalles",
```
[the remark this answers is not in the document]
Comment (diegoAndres01) [reply src/components/data/TableExtraordinaryPayment/Actions/index.tsx:29]: resolved

File: src/components/data/TableExtraordinaryPayment/Actions/styles.ts (hunk 1 above)
Comment (diegoAndres01) [reply src/components/data/TableExtraordinaryPayment/Actions/styles.ts:32]: resolved

File: src/components/data/TableExtraordinaryPayment/Actions/type.ts (hunk 2 above)
Comment (diegoAndres01) [reply src/components/data/TableExtraordinaryPayment/Actions/type.ts:8]: resolved

File: src/components/data/TableExtraordinaryPayment/Detail/index.tsx (hunk 3 above)
Comment (diegoAndres01) [reply src/components/data/TableExtraordinaryPayment/Detail/index.tsx:27]: resolved

File: src/components/modals/ExtraordinaryPaymentModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+export interface ExtraordinaryPaymentModalProps {
+  dataTable: IRowExtraordinaryPayment[];
+  handleClose: () => void;
+  portalId: string;
+  onClickDetails: (id: string) => void;
+  onClickEdit: (id: string) => void;
+  onClickEliminate: (id: string) => void;
+}
+export const ExtraordinaryPaymentModal = (
+  props: ExtraordinaryPaymentModalProps
```
[the remark this answers is not in the document]
Comment (diegoAndres01) [reply src/components/modals/ExtraordinaryPaymentModal/index.tsx]: resolved

**Pull request merged**